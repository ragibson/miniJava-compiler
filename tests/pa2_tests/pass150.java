// PA1 parse classdecls pass
class MainClass {
    public static void main (String [] args) {}
}

class OfItsOwn {
   int A_01;
}  // class OfItsOwn

