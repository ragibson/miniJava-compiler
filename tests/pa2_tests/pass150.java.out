********** miniJava Compiler (PA2) **********
Syntactic Analysis ... 
Syntactic Analysis OK
======= AST Display =========================
Package
  ClassDeclList [2]
  . ClassDecl
  .   "MainClass" classname
  .   FieldDeclList [0]
  .   MethodDeclList [1]
  .   . (public static) MethodDecl
  .   .   VOID BaseType
  .   .   "main" methodname
  .   .   ParameterDeclList [1]
  .   .   . ParameterDecl
  .   .   .   ArrayType
  .   .   .     ClassType
  .   .   .       "String" Identifier
  .   .   .   "args"parametername 
  .   .   StmtList [0]
  . ClassDecl
  .   "OfItsOwn" classname
  .   FieldDeclList [1]
  .   . (public) FieldDecl
  .   .   INT BaseType
  .   .   "A_01" fieldname
  .   MethodDeclList [0]
=============================================
